File: core/src/test/java/javabot/commands/ConfigureTest.java
package javabot.commands;

import com.google.inject.Inject;
import javabot.BaseMessagingTest;
import javabot.dao.ConfigDao;
import javabot.model.Config;
import org.testng.Assert;
import org.testng.annotations.Test;

import static java.lang.String.format;


@Test
public class ConfigureTest extends BaseMessagingTest {
    @Inject
    public ConfigDao configDao;

    public void change() {
        Config config = configDao.get();
        Integer throttleThreshold = config.getThrottleThreshold();
        Assert.assertNotNull(throttleThreshold);

        testMessage("~admin configure", config.toString());

        testMessage("~admin configure --property=throttleThreshold --value=15", format("Setting %s to %d", "throttleThreshold", 15));

        Assert.assertEquals(configDao.get().getThrottleThreshold(), new Integer(15));

        testMessage("~admin configure --property=throttleThreshold --value=10", format("Setting %s to %d", "throttleThreshold", 10));

        Assert.assertEquals(configDao.get().getThrottleThreshold(), new Integer(10));
    }
}

File: src/test/kotlin/javabot/commands/ConfigureTest.kt
package javabot.commands

import com.google.inject.Inject
import javabot.BaseTest
import javabot.dao.ConfigDao
import org.testng.Assert
import org.testng.annotations.Test
import java.lang.String.format


@Test
class ConfigureTest : BaseTest() {
    @Inject
    private lateinit var configDao: ConfigDao
    @Inject
    private lateinit var operation: Configure

    fun change() {
        val config = configDao.get()
        val throttleThreshold = config.throttleThreshold
        Assert.assertNotNull(throttleThreshold)

        var response = operation.handleMessage(message("~admin configure"))
        Assert.assertEquals(response[0].value, config.toString())

        response = operation.handleMessage(message("~admin configure --property=throttleThreshold --value=15"))
        Assert.assertEquals(response[0].value, format("Setting %s to %d", "throttleThreshold", 15))

        Assert.assertEquals(configDao.get().throttleThreshold, 15)

        response = operation.handleMessage(message("~admin configure --property=throttleThreshold --value=10"))
        Assert.assertEquals(response[0].value, format("Setting %s to %d", "throttleThreshold", 10))

        Assert.assertEquals(configDao.get().throttleThreshold, 10)
    }
}